trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/CExpressionTranslator.java

```diff
@@ -588,7 +589,14 @@ public Result handlePostfixIncrementAndDecrement(final ILocation loc, final int
 		builder.setOrResetLrValue(new RValue(valueXcremented, oType, false, false));
 		final ExpressionResult assign = funMakeAssignment.apply(builder.build());
 
-		return new ExpressionResultBuilder().addAllExceptLrValue(assign).setLrValue(tmpRValue).build();
+		final ExpressionResultBuilder resultBuilder = new ExpressionResultBuilder().setLrValue(tmpRValue);
+		if (oType.isAtomic()) {
+			final AtomicStatement atomic = new AtomicStatement(loc, assign.getStatements().toArray(Statement[]::new));
```

Comment:
The atomic block contains all statements inside the original `exprRes`. Could this not be unsound? Only the assignment of `auxvar` and the assignment created by `funMakeAssignment` should be atomic.

Reply:
> The atomic block contains all statements inside the original `exprRes`. Could this not be unsound?

Yes, you are right in principle. For example, if we have some variable `x` that is a pointer to `_Atomic int`, then the pointer increment in `*(x++)++` 😈 should not be executed atomically. 

> Only the assignment of `auxvar` and the assignment created by `funMakeAssignment` should be atomic.

Also the "actual" `switchToRValue` (i.e., the pointer read, without the original `ExpressionResult`) has to be executed atomically. But I currently don't how to handle this properly, without refactoring of every usage `switchToRValue` and `makeAssignment`...

Reply:
Should be fixed in [a54bfac](https://github.com/ultimate-pa/ultimate/pull/688/commits/a54bfacabb660ea591a5e7f8d0839fc5b3e1d060)